### DISCUSSION TITLE
Change of node position when end action of drag

### DISCUSSION THREAD

**Discussion by josielqogit** [xyflow/xyflow#3351] [General]
In this  [example](https://reactflow.dev/docs/examples/interaction/drag-and-drop/)  , when I release the node instead of staying where I released it, it changes its horizontal position to the right. Is there any setting that forces it to stay exactly where I dropped it? When the first node is dropped onto the canvas it is also moved to the center of the canvas. The next nodes remain where they were released, shifting to the right as already mentioned. I would like the first node to also stay where we left it, without this displacement. it's possible?

**Comment by bcakmakoglu** [comment]
You would have to subtract the width/height of the node from the position during drop (which isn't done in the example, probably because a node you use, for example a custom one, could have different heights and widths and not a fixed one).
If you use dynamic width/height you would first have to drop the node, then update the position after the drop again.

**Reply by moklick** [reply]
or you can set nodeOrigin = [0.5, 0.5] to change the node anchor point https://reactflow.dev/docs/api/react-flow-props/#nodeorigin

**Reply by bcakmakoglu** [reply]
I always forget that option exists 😂

**Comment by josielqogit** [comment]
Changing the origin of the node to the center behaved exactly as I wanted. What about the issue of the first node being redirected to the center of the screen no matter where I drop it? Is there any configuration regarding this or would I have to do it manually, repositioning the node after it is released?
